global_logger.py

```diff
@@ -1,23 +1,174 @@
-import logging
 import logging.handlers
 import os
+import time
+from enum import auto, Enum
+from functools import wraps
+from typing import Optional
 
+# Set up logging level constants
+CRITICAL = logging.CRITICAL  # 50
+ERROR = logging.ERROR  # 40
+WARNING = logging.WARNING  # 30
+INFO = logging.INFO  # 20
 VERBOSE = 15
-verbose_level = 'VERBOSE'
-log_file = './logs/logon_server.log'
+DEBUG = logging.DEBUG  # 10
 
-log_formatter = logging.Formatter("%(levelname)-7.7s %(process)-12.12d %(processName)-12.12s %(threadName)-12.12s %(asctime)s: %(message)s")
+# Setup internal vars
+_verbose_level_name = 'VERBOSE'
+_log_file = './logs/logon_server.log'
+
+# noinspection SpellCheckingInspection
+_log_formatter = logging.Formatter(
+    "%(levelname)-7.7s %(process)-12.12d %(processName)-12.12s %(threadName)-12.12s %(asctime)s: %(message)s")
 logger = logging.getLogger()
-logging.addLevelName(VERBOSE, verbose_level)
+logging.addLevelName(VERBOSE, _verbose_level_name)
+
+# Make directory (if it doesn't exist)
+os.makedirs(os.path.dirname(_log_file), exist_ok=True)
```

Comment:
I added this mainly because I wanted to run unit tests on my local copy without manually creating the log file when it didn't exist.